scalopus_tracing/src/native/tracepoint_collector_native.h

```diff
@@ -116,25 +117,17 @@ using NamedCounter = std::tuple<std::string, unsigned int>;
 
 /**
  * @brief A singleton class that keeps track of the ringbuffer allocated to each thread to insert tracepoints into.
+ *        If a thread goes out of scope the ringbuffer for that thread goes into the orphaned collection. This ensures
+ *        that no events get lost when a thread goes out of scope before its events are exfiltrated by the endpoint
+ *        native trace sender.
  */
-class TracePointCollectorNative : public MapTracker<unsigned long, tracepoint_collector_types::ScopeBufferPtr>
+class TracePointCollectorNative
 {
-private:
-  TracePointCollectorNative() = default;
-  TracePointCollectorNative(const TracePointCollectorNative&) = delete;
-  TracePointCollectorNative& operator=(const TracePointCollectorNative&) = delete;
-  TracePointCollectorNative& operator=(TracePointCollectorNative&&) = delete;
-
-  /**
-   * @brief The size of each thread's ringbuffer.
-   * If this is too small, and the thread produces events quicker than the server thread collects them this will result
-   * in lost events.
-   */
-  std::size_t ringbuffer_size_{ 10000 };
-
 public:
   using Ptr = std::shared_ptr<TracePointCollectorNative>;
   using WeakPtr = std::weak_ptr<TracePointCollectorNative>;
+  using BufferMap = MapTracker<unsigned long, tracepoint_collector_types::ScopeBufferPtr>;
+  using BufferVector = std::vector<std::pair<unsigned long, tracepoint_collector_types::ScopeBufferPtr>>;
```

Comment:
This is deliberately a vector of pairs instead of a map, if you join a thread and then make a new one `pthread_self()` returns;
```
// make new thread, print pthread_self():
140581527353088
// join thread
// make new thread, print pthread_self():
140581527353088
```

So thread id's are reused.